pkg/flow/internal/controller/node_config_logging.go

```diff
@@ -80,3 +81,19 @@ func (cn *LoggingConfigNode) Block() *ast.BlockStmt {
 
 // NodeID implements dag.Node and returns the unique ID for the config node.
 func (cn *LoggingConfigNode) NodeID() string { return cn.nodeID }
+
+// UpdateBlock updates the River block used to construct arguments.
+// The new block isn't used until the next time Evaluate is invoked.
+//
+// UpdateBlock will panic if the block does not match the component ID of the
+// LoggingConfigNode.
+func (cn *LoggingConfigNode) UpdateBlock(b *ast.BlockStmt) {
+	if !BlockComponentID(b).Equals(strings.Split(cn.nodeID, ".")) {
+		panic("UpdateBlock called with an River block with a different ID")
+	}
+
```

Comment:
if https://github.com/grafana/agent/pull/5992 is merged first, we might need to re-create the buffer here